std/algorithm/iteration.d

```diff
@@ -4559,10 +4559,10 @@ if (isInputRange!R && !isInfinite!R)
     {
         foreach (k; 0 .. data.length / 16)
         {
-            static if (isRandomAccessRange!R && hasSlicing!R)
+            static if (isRandomAccessRange!R && hasLength!R && hasSlicing!R)
             {
                 store[idx] = sumPairwise16!F(data);
-                data = data[16 .. $];
```

Comment:
> $ in slices should be defined by the compiler for all arrays

It's not guaranteed that a random access range is an array, even if it's not infinite and has slicing. Take a look at the result of `std.conv.toChars` for example, who's underlying data is a number.